### ISSUE TITLE
CONTENT004 should not infer document style from an ambiguous first heading

### ISSUE THREAD

**Issue by joshrotenberg** [joshrotenberg/mdbook-lint#462]
## Summary

With `style = "consistent"`, CONTENT004 fixes the expected style from the first multi-word heading. If that heading is compatible with both title case and sentence case, the detector silently chooses sentence case and then flags an otherwise consistently Title Case document.

## Reproduction

```markdown
# Agentic SDLC

## Installation Steps

## Configuration Options
```

Current behavior: the first heading is considered ambiguous because `SDLC` is skipped as an acronym. It becomes sentence case, and both subsequent headings are reported.

[jmbowman/portabase documents the same production failure](https://github.com/jmbowman/portabase/blob/8683673866cb20411b1836f5f52c3a5be593538b/docs/wiki/mdbook-lint-bugs.md#3-content004-defaults-to-sentence-case-for-shortambiguous-headings-overriding-an-otherwise-title-case-document).

## Root cause

[`detect_style`](https://github.com/joshrotenberg/mdbook-lint/blob/8775228dda9db010cc0083c2af54ada98101f53c/crates/mdbook-lint-rulesets/src/content/content004.rs#L170-L181) maps every ambiguous result (`is_title && is_sentence`) to `SentenceCase`. The check then stores that first result as the document-wide baseline.

## Expected behavior

Ambiguous headings should not decide the document style. Reasonable options include:

- return an `Ambiguous` result and wait for the first discriminating heading;
- infer from multiple headings and choose the dominant style;
- allow a configurable tie-breaker, while keeping explicit `style = "title"` / `"sentence"` unchanged.

## Acceptance tests

- `Agentic SDLC` followed by Title Case headings yields no violations in consistent mode.
- An ambiguous heading followed by sentence-case headings also yields no violations.
- Once a discriminating style is established, genuinely inconsistent headings are still reported.
